sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonUtils.scala

```diff
@@ -55,4 +60,26 @@ object JacksonUtils {
 
     schema.foreach(field => verifyType(field.name, field.dataType))
   }
+
+  private def validateStringLiteral(exp: Expression): String = exp match {
+    case Literal(s, StringType) => s.toString
+    case e => throw new AnalysisException("Must be a string literal, but: " + e)
+  }
+
+  def validateSchemaLiteral(exp: Expression): StructType =
+    DataType.fromJson(validateStringLiteral(exp)).asInstanceOf[StructType]
```

Comment:
I guess we should check if it is `StructType` and throw a proper exception because it seems `JsonToStruct` does not check if `exp` is `StructType` and it probably throws a cast exception (if I haven't missed something here).